hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/FormatUtils.java

```diff
@@ -174,82 +168,34 @@ public static HoodieMergedLogRecordScanner logScanner(
-
-      this.executor = new BoundedInMemoryExecutor<>(
-          StreamerUtil.getMaxCompactionMemoryInBytes(flinkConf),
-          getParallelProducers(scannerBuilder),
-          Option.empty(),
-          Function.identity(),
-          new DefaultSizeEstimator<>(),
-          Functions.noop());
-      this.iterator = this.executor.getRecordIterator();
-
-      // Start reading and buffering
-      this.executor.startProducingAsync();
-    }
-
-    public Iterator<HoodieRecord<?>> getRecordsIterator() {
-      return this.iterator;
-    }
-
-    /**
-     * Setup log and parquet reading in parallel. Both write to central buffer.
-     */
-    private List<HoodieProducer<HoodieRecord<?>>> getParallelProducers(
-        HoodieUnMergedLogRecordScanner.Builder scannerBuilder
-    ) {
-      List<HoodieProducer<HoodieRecord<?>>> producers = new ArrayList<>();
-      producers.add(new FunctionBasedQueueProducer<>(queue -> {
-        HoodieUnMergedLogRecordScanner scanner =
-            scannerBuilder.withLogRecordScannerCallback(queue::insertRecord).build();
-        // Scan all the delta-log files, filling in the queue
-        scanner.scan();
-        return null;
-      }));
+  public static HoodieUnMergedLogRecordScanner getUnMergedLogRecordScanner(MergeOnReadInputSplit split,
+                                                                           Schema logSchema,
+                                                                           InternalSchema internalSchema,
+                                                                           Configuration hadoopConf,
+                                                                           org.apache.flink.configuration.Configuration flinkConf) {
+    List<String> mergers = Arrays.stream(flinkConf.getString(FlinkOptions.RECORD_MERGER_IMPLS).split(","))
+        .map(String::trim)
+        .distinct()
+        .collect(Collectors.toList());
 
-      return producers;
-    }
+    HoodieRecordMerger merger = HoodieRecordUtils.createRecordMerger(
+        split.getTablePath(), EngineType.FLINK, mergers, flinkConf.getString(FlinkOptions.RECORD_MERGER_STRATEGY_ID));
 
-    public void close() {
-      this.executor.shutdownNow();
-    }
+    return HoodieUnMergedLogRecordScanner.newBuilder()
+        .withStorage(HoodieStorageUtils.getStorage(
+            split.getTablePath(), HadoopFSUtils.getStorageConf(hadoopConf)))
+        .withBasePath(split.getTablePath())
+        .withLogFilePaths(split.getLogPaths().get())
+        .withReaderSchema(logSchema)
+        .withInternalSchema(internalSchema)
+        .withLatestInstantTime(split.getLatestCommit())
+        .withReverseReader(false)
+        .withBufferSize(
+            flinkConf.getInteger(HoodieMemoryConfig.MAX_DFS_STREAM_BUFFER_SIZE.key(),
+                HoodieMemoryConfig.DEFAULT_MR_MAX_DFS_STREAM_BUFFER_SIZE))
+        .withInstantRange(split.getInstantRange())
+        .withRecordMerger(merger)
```

Comment:
Not sure whether the merger is necessary because for `HoodieUnMergedLogRecordScanner`, the merge never happens.